dbms/src/Common/ProcessCollector.cpp

```diff
@@ -0,0 +1,48 @@
+// Copyright 2023 PingCAP, Ltd.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <Common/ProcessCollector.h>
+#include <ProcessMetrics/ProcessMetrics.h>
+
+namespace DB
+{
+
+ProcessCollector::ProcessCollector()
+{
+    auto info = get_process_metrics();
+    start_time.Set(info.start_time);
+}
+
+std::vector<prometheus::MetricFamily> ProcessCollector::Collect() const
+{
+    auto new_info = get_process_metrics();
+
+    {
+        std::lock_guard<std::mutex> lock(mu);
```

Comment:
It looks like `std::lock_guard<std::mutex> lock(mu);` is useless since `prometheus::Gauge` is thread safe.